src/main/java/com/dumbdogdiner/stickyapi/common/user/StickyUser.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (c) 2020 DumbDogDiner <dumbdogdiner.com>. All rights reserved.
+ * Licensed under the MIT license, see LICENSE for more information...
+ */
+package com.dumbdogdiner.stickyapi.common.user;
+
+import com.dumbdogdiner.stickyapi.bukkit.user.StickyUserBukkit;
+import com.dumbdogdiner.stickyapi.common.cache.Cacheable;
+import lombok.Getter;
+import org.bukkit.Bukkit;
+import org.bukkit.OfflinePlayer;
+import org.bukkit.entity.Player;
+
+import java.util.UUID;
+
+public class StickyUser implements Cacheable {
```

Comment:
The problem with a class like this is it effectively does nothing. Craftbukkit caches players, and this is doing this too i.e. it will barely be any quicker. Yes, if the server needs to fetch an entire profile, the stock implementation will be slower. However, there are better ways to do this than how it's being done here. We have discussed having plugin APIs before, but I vote this is the best solution. It is much better to exchange what data a plugin needs with another than to extend everything, despite not needing 90% of that information.

Reply:
This is very useful to cache OFFLINE users especially. Every plugin's user should have a few specific methods, namely getters for UUID, and if it's done via inheritance, it allows a plugin api to generically take any other plugin's user and compare stuff.

Reply:
You don't need to do that if you have plugins implement methods that take UUIDs. The plugins can cache their own data on offline users.

Reply:
A lot of the time, retrieving cached offline user data for each plugin will require a database request, there's no real way to get around that. If they're written properly, they'll then cache that themselves.

Reply:
For retrieving just user names and the last login time from bukkit doesn't require a database. A plugin can be written to use one, but it doesn't have to.

Reply:
As far as the root class vs passing UUIDs:
I just feel that this:
```java
StickyWalletUser swu = getTransactionUser(something); // just go with it mkay
StickyCommands.getInstance().isAFK(swu);
```
or
```java
(new StickyCommands.User(getTransactionUser(something))).isAFK();
```

is cleaner than

```java
(new StickyCommands.User(getTransactionUser(something).getUniqueId())).isAFK();
```

Reply:
@aakatz3 `StickyCommands.getInstance().getUser(UUID).isAfk()` (or smth like this) This is a very reasonable way of getting a user object from a different plugin.

`StickyCommands.getInstance().isAFK(swu);` doing this defeats the point of it being an object, why not just store the UUIDs in a hashmap (UUID, Boolean) at that point?